### DISCUSSION TITLE
Moving First Person Camera while in NoClip mode

### DISCUSSION THREAD

**Discussion by Rob-Meadows2001** [pokepetter/ursina#225] [General]
When using FirstPersonController I can move the mouse to move the players' camera view. However, when I tried to implement noClip mode (using player.add_script(NoclipMode(speed=20, require_key='shift')) ), I can enter no clip mode and move in all 6 directions but I can't move the camera around with the mouse as it seems to be locked in place while holding shift. How do I fix this?

I've attached my source code for reference. (I'm using python 3.9 with the Pycharm IDE)

`from ursina import *
from ursina.prefabs.first_person_controller import FirstPersonController

class Voxel(Button):
    def __init__(self, position=(0, 0, 0), height=0):
        super().__init__(
            parent=scene,
            position=position,
            model="cube",
            origin=Vec3(5, 5, 5),
            texture="white_cube",
            color=color.Color(1, 1, 1, 0.1),
            highlight_color=color.Color(1, 0, 0, 0.1),
        )

app = Ursina()

window.title = "Powered Box Puzzle - Solution"
window.borderless = True
window.fullscreen = True
window.exit_button.visible = False
window.fps_counter.enabled = True

side = 9
layer = 9
for z in range(side):
    for y in range(layer):
        for x in range(side):
            voxel = Voxel(position=(x, y, z), height=layer)

player = FirstPersonController(speed=10, mouse_sensitivity=Vec2(50, 50))
player.add_script(NoclipMode(speed=20, require_key='shift'))
app.run()
`

**Comment by pokepetter** [comment]
By making it yourself.

**Reply by Rob-Meadows2001** [reply]
Do you mean it isn't already a feature?
